service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -193,7 +194,17 @@ void testUpdateMetadataWithInvalidUser() {
   @Test
   void testUpdateMetadata() {
     mockDataset();
-    String metadata = "test metadata";
+    String metadata = """
+        {"test": "metadata"}""";
+    when(samService.hasGlobalAction(user, SamAction.UPDATE_ANY_METADATA)).thenReturn(true);
+    datasetService.updateMetadata(user, datasetId, metadata);
+    verify(datasetDao).update(dataset.withMetadata(metadata));
+  }
+
+  @Test(expected = BadRequestException.class)
```

Comment:
I don't think JUnit's `expected` is a good way to detect an expected exception, since there's no way to verify which statement produced the exception. Instead I'd assert on the `updateMetadata` call.
```
    assertThrows(BadRequestException.class, () ->  datasetService.updateMetadata(user, datasetId, metadata));
```